### ISSUE TITLE
execp + execbar is broken

### ISSUE THREAD

**Issue by j4nu5** [brndnmtthws/conky#802]
@lasers @su8 I think #646 does not really solve the problem and in fact causes another problem:

I have a dynamic list of execbars:

```
${exec tail -n1 /tmp/build-logs/*.status | egrep "^[0-9]+$" | awk '{print "${execbar echo " $0 "}"}'}
${execp tail -n1 /tmp/build-logs/*.status | egrep "^[0-9]+$" | awk '{print "${execbar echo " $0 "}"}'}
```

(The `exec` line is for referencing the real value for the bar for debugging).

#646 Caching the result of execbars makes them display incorrect progress information. Specifically, some of the progress bars use previous values from other progress bars. Attaching a gif for reference.

![conky](https://user-images.githubusercontent.com/992799/53688175-099d0f00-3d40-11e9-8d9d-427ec020389c.gif)

@lasers Requesting to re-open the issue.

_Originally posted by @j4nu5 in https://github.com/brndnmtthws/conky/issues/580#issuecomment-468965537_

**Comment by j4nu5** [comment]
Continuing from https://github.com/brndnmtthws/conky/issues/580#issuecomment-468968217 

@lasers 
"To be sure, you can try reverting that commit (or drop the cache code)... to see it's still happening." - Yes, created a local build with a revert of a644600bc5ae8bf0971cdaf992b7c1d6c7c92553 and can see:

```
conky: reading exec value failed (perhaps it's not the correct format?)
```

with your test config:
```
conky.config = {
    out_to_x=false,
    out_to_console=true,
    own_window=true,
    update_interval=1,
}
conky.text = [[
    ${execp python -c 'from random import randint;print("BAR: ${execbar echo %s}" % randint(10,99))'}
    ${execp python -c 'from random import randint;print("BAR: ${execbar echo %s}" % randint(11,99))'}
    ${execp python -c 'from random import randint;print("BAR: ${execbar echo %s}" % randint(12,99))'}
]]
```

**Comment by lasers** [comment]
Despite lot of `reading exec value failed`, it sometimes catch numbers. I'm not sure why.

You'll have to wait for someone/volunteers/yourself to look into this and solve the issue. I don't know C/C++. I am collaborator to confirm bugs, review stale issues, find duplicates, close issues, etc.

This is still happening with https://github.com/brndnmtthws/conky/issues/580#issuecomment-411788147.

**Comment by j4nu5** [comment]
@lasers Happy to give it a shot. Just a few clarifications needed:

- Are you OK with `Google LLC` being listed in the AUTHORS file?
- I am assuming the project is GPLv3 licensed? I am confused with the presence of 2 license files: GPLv3 and BSD in the repo root?

If the answer to both questions is yes, please assign the issue to me :)

**Comment by lasers** [comment]
@brndnmtthws Not my place. Can you please answer the questions? :)

**Comment by brndnmtthws** [comment]
@j4nu5 AUTHORS isn't kept up to date anymore, it's just there for historical reasons. Feel free to add to it if you want. Conky was originally BSD, but we switched to GPLv3 several years ago. Everything since then is GPLv3.

**Comment by j4nu5** [comment]
Got it, thanks!

Update: Seems like its not limited to execbar but exec inside execp is broken?
```
conky.config = {
    out_to_x=false,
    out_to_console=true,
    own_window=true,
    update_interval=0.2,
}
conky.text = [[
    ${execp python -c 'from random import randint;print("BAR: ${exec echo %s}" % randint(10,99))'}
]]
```

Prints:
```
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 96
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 
    BAR: 41
    BAR: 
    BAR:
```

**Comment by lasers** [comment]
Closed via #804.
